contracts/protocore/OriginProtocore.sol

```diff
@@ -133,6 +133,34 @@ contract OriginProtocore is MasterCopyNonUpgradable, GenesisOriginProtocore, Pro
         );
     }
 
+    /**
+     * @notice It registers a vote for a link specified by vote message hash.
+     *         Validators sign a vote message and provides the signature.
+     *
+     * @dev It must satify pre and post conditions of
+     *      Protocore::registerVoteInternal.
+     *
+     * @param _voteMessageHash Message hash of a vote.
+     * @param _r First 64-bytes of the signature i.e (0,63)
```

Comment:
@param _r --- It's bytes32 ... hence it's the first 32 bytes of the ECDSA signature.
@param _s --- It's bytes32 ... hence it's the second 32 bytes of the ECDSA signature.
@param _v --- It's recover id/value of ECDSA signature.

Reply:
Done.